diesel/src/mysql/connection/url.rs

```diff
@@ -32,9 +33,13 @@ impl ConnectionOptions {
             Some(host) => Some(try!(CString::new(host.as_bytes()))),
             None => None,
         };
-        let user = try!(CString::new(url.username().as_bytes()));
+        let username = percent_decode(url.username().as_ref()).decode_utf8_lossy();
```

Comment:
Is lossy the correct behavior here? If there's an invalid UTF-8 character, I think that should be considered an error condition.

Reply:
Thanks, make sense, updated